### DISCUSSION TITLE
Line plot varying thickness and/or color?

### DISCUSSION THREAD

**Discussion by clo-yunhee** [epezent/implot#653] [Q&A]
Is there a way to make a line plot that can vary in thickness and/or color with ImPlot as-is?
The idea is to represent confidence with a variation in line thickness or color at each point rather than with a shaded plot
Otherwise do you have any pointers to how I could implement something like that with the internal API?

A potential idea I had was to make something like a sparse heatmap but it seems really overkill and I'm not sure if it would be that much better than making a custom line plot at that point

**Comment by keltonhalbert** [comment]
I haven't seen anything regarding varying line thicknesses, but I did port the code from #332 to a local branch of mine for varying color. You're welcome to take a look and see if it helps: https://github.com/keltonhalbert/implot/tree/setcolors.

Presumably, a similar approach could be used for the line primitives to do varying thicknesses.

**Comment by brenocq** [comment]
Hi @clo-yunhee @keltonhalbert! I would love to get your input in #519, we're planning to have a new API to plot lines/points, which should support per-point color (and maybe also per-point thickness in the future?). This will be a major change and (hopefully) huge quality of life improvement, so we're trying to get more feedback on it :)

**Reply by keltonhalbert** [reply]
Hi @brenocq! 

I am definitely in favor of this change, as supporting varying colors and line thicknesses would be beneficial for my use cases. I can check out the PR into my codebase and start messing around with it again. Is there anything in particular you'd like feedback on? I've messed with the variable line colors before and had success getting it to work.

**Reply by brenocq** [reply]
>  I can check out the PR into my codebase and start messing around with it again. Is there anything in particular you'd like feedback on?

That's awesome! I just wanted to know if you like that approach or have some suggestion of improvement. If you have the time to try it, please let me know if you find any bugs or weird behaviors.
